runtime/near-vm-runner/src/tests/runtime_errors.rs

```diff
@@ -604,95 +463,90 @@ fn bad_import_func(env: &str) -> Vec<u8> {
+            ProtocolFeature::FixContractLoadingCost,
+        ])
+        .expects(&[
+            expect![[r#"
+                VMOutcome: balance 4 storage_usage 12 return data None burnt gas 0 used gas 0
+                Err: PrepareError: Error happened during instantiation.
+            "#]],
+            #[cfg(feature = "protocol_feature_fix_contract_loading_cost")]
+            expect![[r#"
+                VMOutcome: balance 4 storage_usage 12 return data None burnt gas 46500213 used gas 46500213
+                Err: PrepareError: Error happened during instantiation.
+            "#]],
+        ]);
 }
 
 #[test]
 fn test_bad_import_2() {
-    with_vm_variants(|vm_kind: VMKind| {
-        let code = bad_import_func("wtf");
-        gas_and_error_match(
-            make_simple_contract_call_vm(&code, "hello", vm_kind),
-            prepaid_loading_gas(code.len()),
-            Some(VMError::FunctionCallError(FunctionCallError::CompilationError(
-                CompilationError::PrepareError(PrepareError::Instantiate),
-            ))),
-        );
-    });
+    test_builder()
+        .wasm(&bad_import_func("wtf"))
+        .method("hello")
+        .protocol_features(&[
+            #[cfg(feature = "protocol_feature_fix_contract_loading_cost")]
+            ProtocolFeature::FixContractLoadingCost,
+        ])
+        .expects(&[
+            expect![[r#"
+                VMOutcome: balance 4 storage_usage 12 return data None burnt gas 0 used gas 0
+                Err: PrepareError: Error happened during instantiation.
+            "#]],
+            #[cfg(feature = "protocol_feature_fix_contract_loading_cost")]
+            expect![[r#"
+                VMOutcome: balance 4 storage_usage 12 return data None burnt gas 45849963 used gas 45849963
+                Err: PrepareError: Error happened during instantiation.
+            "#]],
+        ]);
 }
 
 #[test]
 fn test_bad_import_3() {
-    with_vm_variants(|vm_kind: VMKind| {
-        let msg = match vm_kind {
-            VMKind::Wasmer0 => "link error: Incorrect import type, namespace: env, name: input, expected type: global, found type: function",
-            VMKind::Wasmtime => "\"expected global, but found func\"",
-            VMKind::Wasmer2 => "Error while importing \"env\".\"input\": incompatible import type. Expected Global(GlobalType { ty: I32, mutability: Const }) but received Function(FunctionType { params: [I64], results: [] })",
-        }.to_string();
-        gas_and_error_match(
-            make_simple_contract_call_vm(&bad_import_global("env"), "hello", vm_kind),
-            46500213,
-            Some(VMError::FunctionCallError(FunctionCallError::LinkError { msg })),
-        );
-    });
+    test_builder().wasm(&bad_import_global("env")).method("hello").opaque_error().expect(expect![
+        [r#"
+        VMOutcome: balance 4 storage_usage 12 return data None burnt gas 46500213 used gas 46500213
+        Err: ...
```

Comment:
I wonder if we need something to specify different errors for different VMs. Or maybe we specifically don't care about what the error instance?
I'm just thinking that in the old version of the test, it was obvious from the code that there is a difference and AND if any of the errors changed, we would see it in the test.

Reply:
Yeah, I debated this. We *could* introduce multi-expect functionality here, the same way we do for protocl features. 

However, I don't think we actually care. It's a property of our systems that errors are opaque. So we don't loose a lot if we erase everything except the fact that the error occured. Rather than supporting results between vm being different, I decided to double down on expressing that they should be sufficiently the same. 

There are a couple of cases where the results are different and we care about that, that is hacked around by skipping two of the three runtimes (see `test_nan_sign`).